src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -605,27 +610,39 @@ public boolean isUserSuspended(UserProfileUpdatedData userProfileUpdatedData, St
 
     public boolean isEachRoleUpdated(UserProfileUpdatedData userProfileUpdatedData, String userId,
                                      String origin, long rowId) {
+        boolean isEachRoleUpdated = true;
         try {
             Optional<Object> resultResponse = getUserProfileUpdateResponse(userProfileUpdatedData, userId, origin);
 
-            if (!resultResponse.isPresent()
-                || (isNull(((UserProfileRolesResponse) resultResponse.get())
-                .getRoleAdditionResponse()))
-                || (((UserProfileRolesResponse) resultResponse.get()).getRoleAdditionResponse()
-                .getIdamStatusCode().equals(valueOf(HttpStatus.CREATED.value())) == FALSE)) {
+            if (resultResponse.isPresent() && resultResponse.get() instanceof UserProfileRolesResponse) {
+                UserProfileRolesResponse userProfileRolesResponse = (UserProfileRolesResponse) resultResponse.get();
+                if (nonNull(userProfileRolesResponse.getRoleAdditionResponse())) {
+                    if (isNotTrue(userProfileRolesResponse.getRoleAdditionResponse().getIdamStatusCode()
+                            .equals(valueOf(HttpStatus.CREATED.value())))) {
+
+                        validationServiceFacade.logFailures(String.format(IDAM_STATUS,
+                                userProfileRolesResponse.getRoleAdditionResponse().getIdamMessage()), rowId);
+                        isEachRoleUpdated = false;
+                    }
+                } else {
+                    validationServiceFacade.logFailures(UP_FAILURE_ROLES, rowId);
+                    isEachRoleUpdated = false;
+                }
+            } else {
                 validationServiceFacade.logFailures(UP_FAILURE_ROLES, rowId);
-                return false;
+                isEachRoleUpdated = false;
             }
 
         } catch (Exception ex) {
             log.error("{}:: UserProfile modify api failed for row ID {} with error :: {}:: Job Id {}",
-                loggingComponentName, rowId, ex.getMessage(), validationServiceFacade.getAuditJobId());
+                    loggingComponentName, rowId, ex.getMessage(), validationServiceFacade.getAuditJobId());
             validationServiceFacade.logFailures(UP_FAILURE_ROLES, rowId);
-            return false;
+            isEachRoleUpdated = false;
         }
-        return true;
+        return isEachRoleUpdated;
     }
 
+
```

Comment:
I had to add multiple Else blocks to cover all possible conditions and prevent NPE's. If there is any possibility of removing them based on realtime values, please suggest.